indico/modules/events/abstracts/forms.py

```diff
@@ -452,12 +453,26 @@ class EditEmailTemplateTextForm(IndicoForm):
 
     def __init__(self, *args, **kwargs):
         self.event = kwargs.pop('event')
+        self.email_tpl = kwargs['obj']
         super().__init__(*args, **kwargs)
         choices = [('', config.NO_REPLY_EMAIL)]
         choices += list(self.event.get_allowed_sender_emails(extra=self.reply_to_address.object_data).items())
         self.reply_to_address.choices = choices
         self.body.description = render_placeholder_info('abstract-notification-email', event=self.event)
 
+    def validate_body(self, field):
+        # disallow using the invitation_url placeholder for non-invited email templates
+        has_invited_rule = any(
+            AbstractState.invited.value in value
+            for rule in self.email_tpl.rules
+            for value in rule.values()
+        )
+        if not has_invited_rule and AbstractInvitationURLPlaceholder.is_in(
+            field.data, abstract=None
+        ):
```

Comment:
That fits in one line, which looks much better as well ;)

Also, the logic needs to be `has_invited_rule != AbstractInvitationURLPlaceholder.is_in(field.data, abstract=None)` with a modified error message.

Or even better, having two checks (`has_invited_rule and not has_placeholder` and `not has_invited_rule and has_placeholder`) possibly with two separate checks to make the message less awkward:

- "Only invitation email templates may contain the ... placeholder"
- "Invitation email templates must contain the ... placeholder"